src/modules/gfx_mandelbrot.c

```diff
@@ -59,47 +65,51 @@ void reset(void) {
 	nexttick = udate();
 	matrix_clear();
 	frame = 0;
+	min = 0;
+	max = 0;
 }
 
-int draw(int argc, char* argv[]) {
-	int min;
-	int max;
-
+void drawrow(void* row) {
+	int py = *(int*) row;
 	int px;
-	int py;
-
-	for (py = 0; py < my; py++)
-		for (px = 0; px < mx; px++) {
-			float zoom = SCALE(frame, FRAMES) * MAXZOOM;
-			float x0 = ((SCALE(px, mx) * 3.5f) - 2.5f) / zoom;
-			float y0 = ((SCALE(py, my) * 2.0f) - 1.0f) / zoom;
-
-			float x = 0;
-			float y = 0;
-			int i = 0;
-			while ((x*x + y*y) <= (2*2) && i < ITERATIONS) {
-				float xt = x*x - y*y + x0;
-				y = 2 * x * y + y0;
-				x = xt;
-				i++;
-				if (i < min) min = i;
-				if (i > max) max = i;
-			}
-
-			iters[PPOS(px, py)] = i;
+	for (px = 0; px < mx; px++) {
+		float zoom = SCALE(frame, FRAMES) * MAXZOOM;
+		float x0 = ((SCALE(px, mx) * 3.5f) - 2.5f) / zoom;
+		float y0 = ((SCALE(py, my) * 2.0f) - 1.0f) / zoom;
+
+		float x = 0;
+		float y = 0;
+		int i = 0;
+		while ((x*x + y*y) <= (2*2) && i < ITERATIONS) {
+			float xt = x*x - y*y + x0;
+			y = 2 * x * y + y0;
+			x = xt;
+			i++;
+			oscore_mutex_lock(lock);
+			if (i < min) min = i;
+			if (i > max) max = i;
+			oscore_mutex_unlock(lock);
 		}
 
-	for (py = 0; py < my; py++)
-		for (px = 0; px < mx; px++) {
-			int i = iters[PPOS(px, py)];
-			RGB col = RGB(0, 0, 0);
-			if (i != (ITERATIONS)) {
-				if (min == max) max = min + 1;
-				byte scaled = rescale(i - min, max, 255);
-				col = HSV2RGB(HSV(scaled, 255, 255));
-			}
-			matrix_set(px, py, &col);
+		RGB col = RGB(0, 0, 0);
+		if (i != (ITERATIONS)) {
+			oscore_mutex_lock(lock);
+			if (min == max) max = min + 1;
+			int mmin = min;
+			int mmax = max;
+			oscore_mutex_unlock(lock);
+			byte scaled = rescale(i - mmin, mmax, 255);
+			col = HSV2RGB(HSV(scaled, 255, 255));
 		}
+		matrix_set(px, py, &col);
+	}
+
+}
+
+int draw(int argc, char* argv[]) {
+	int py;
+
+	taskpool_forloop(TP_GLOBAL, &drawrow, 0, my);
```

Comment:
This SEGVs beautifully on macOS! (compiled with clang-902.0.39.2)

```
>> Now drawing mandelbrotASAN:DEADLYSIGNAL
=================================================================
==9002==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x00010eac395e bp 0x7ffee1141ef0 sp 0x7ffee1141ef0 T0)
==9002==The signal is caused by a READ memory access.
==9002==Hint: address points to the zero page.
    #0 0x10eac395d in taskpool_submit taskpool.c:119
    #1 0x10eac3cfd in taskpool_submit_array taskpool.c:135
    #2 0x10eac3dba in taskpool_forloop taskpool.c:151
    #3 0x1122d975e in draw (gfx_mandelbrot.so:x86_64+0x175e)
    #4 0x10eabecc4 in sled_main main.c:282
    #5 0x7fff7a5b5014 in start (libdyld.dylib:x86_64+0x1014)

==9002==Register values:
rax = 0x0000000000000000  rbx = 0x0000611000054500  rcx = 0x0000100000000000  rdx = 0x0000611000054500  
rdi = 0x0000000000000000  rsi = 0x00000001122d93b0  rbp = 0x00007ffee1141ef0  rsp = 0x00007ffee1141ef0  
 r8 = 0x0000000000000004   r9 = 0x0000000000000000  r10 = 0xffffffffffffffff  r11 = 0x00000fffffffffff  
r12 = 0x0000000000000000  r13 = 0x0000000000000040  r14 = 0x0000000000000004  r15 = 0x00000001122d93b0  
AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV taskpool.c:119 in taskpool_submit
==9002==ABORTING
[1]    9002 abort      ./sled
```

So evidently, TP_GLOBAL is NULL!

Reply:
Why that is, I have no idea though.

Reply:
I don't have a macOS machine available right now.

Add some asserts in the core of sled?

Reply:
An assert(TP_GLOBAL != NULL) in sled_main right after taskpool_create goes through, of course.

`Assertion failed: (TP_GLOBAL != NULL), function init, file src/modules/gfx_mandelbrot.c, line 51.`

This, however, doesn't. This might be related to linking??, but I'm not too proficient in that area - is there anything i should/could check?

Reply:
No idea. It should be accessible just like the timer global.
Maybe I forgot a modifier?

Reply:
I think I know how to fix this case.
It's not pretty, but it'll work.